src/display/api.js

```diff
@@ -2127,6 +2134,13 @@ class WorkerTransport {
 
       sink.onCancel = reason => {
         rangeReader.cancel(reason);
+
+        sink.ready.catch(readyReason => {
+          if (this.destroyed) {
+            return; // Ignore any pending requests if the worker was terminated.
+          }
+          throw readyReason;
+        });
```

Comment:
Compared to the `fullReader` case above, I wasn't actually able to trigger this code-path in any of the unit-tests. However, given the (fairly) small amount of code, the fact that this is consistent with the above, and to prevent any future issues I figured that this cannot hurt.

Given the (relatively) small size of a range request, as compared to the full request, triggering this is probably a lot more difficult and timing-dependent. E.g. it *may* still happen for a slow connection and/or a large `rangeChunkSize` option.